services/streammanager/lambda/lambdamanager.go

```diff
@@ -0,0 +1,92 @@
+//go:generate mockgen --build_flags=--mod=mod -destination=../../../mocks/services/streammanager/lambda/mock_lambda.go -package mock_lambda github.com/rudderlabs/rudder-server/services/streammanager/lambda LambdaClient
+
+package lambda
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/aws/aws-sdk-go/service/lambda"
+	"github.com/mitchellh/mapstructure"
+	"github.com/rudderlabs/rudder-server/services/streammanager/common"
+	"github.com/rudderlabs/rudder-server/utils/awsutils"
+	"github.com/rudderlabs/rudder-server/utils/logger"
+	"github.com/tidwall/gjson"
+)
+
+// Config is the config that is required to send data to Lambda
+type Config struct {
+	ClientContext string
+	Lambda        string
+}
+
+type LambdaProducer struct {
+	client LambdaClient
+}
+
+type LambdaClient interface {
+	Invoke(input *lambda.InvokeInput) (*lambda.InvokeOutput, error)
+}
+
+var pkgLogger logger.LoggerI
+
+func init() {
+	pkgLogger = logger.NewLogger().Child("streammanager").Child(lambda.ServiceName)
+}
+
+// NewProducer creates a producer based on destination config
+func NewProducer(destinationConfig map[string]interface{}, o common.Opts) (*LambdaProducer, error) {
+	sessionConfig, err := awsutils.NewSessionConfig(destinationConfig, o.Timeout, lambda.ServiceName)
+	if err != nil {
+		return nil, err
+	}
+	if sessionConfig.Region == "" {
+		return nil, errors.New("could not find region configuration")
+	}
+	return &LambdaProducer{client: lambda.New(awsutils.CreateSession(sessionConfig))}, nil
+}
+
+// Produce creates a producer and send data to Lambda.
+func (producer *LambdaProducer) Produce(jsonData json.RawMessage, destConfig interface{}) (int, string, string) {
+	client := producer.client
+	if client == nil {
+		return simpleErrorResponse("Could not create client")
+	}
+	data := gjson.GetBytes(jsonData, "payload").String()
+	if data == "" {
+		return simpleErrorResponse("Invalid payload")
+	}
+
+	var config Config
+	err := mapstructure.Decode(destConfig, &config)
+	if err != nil {
+		return simpleErrorResponse("error while decoding destination config :: " + err.Error())
+	}
+
+	var invokeInput lambda.InvokeInput
+	invokeInput.SetFunctionName(config.Lambda)
+	invokeInput.SetPayload([]byte(data))
+	invokeInput.SetInvocationType("Event") // Setting InvocationType as Event(Asynchronous Invocation)
+	if config.ClientContext != "" {
+		invokeInput.SetClientContext(config.ClientContext)
+	}
+
+	if err = invokeInput.Validate(); err != nil {
+		return simpleErrorResponse("Invalid invokeInput :: " + err.Error())
+	}
+
+	_, err = client.Invoke(&invokeInput)
```

Comment:
Isn't it necessary to verify that the invocation was indeed successful by looking into the returned `lambda.InvokeOutput` `FunctionError` and `StatusCode` fields?

Reply:
Among the two possible type of errors, Invocation errors & function runtime errors, Invocation error will always have a non nil err which is being handled.
As we've decided to use only asynchronous invocation, lambda won't be returning any runtime errors in the response.